homeassistant/components/elmax/common.py

```diff
@@ -0,0 +1,229 @@
+            raise HomeAssistantError("An unexpected error occurred") from err
+
+
+class ElmaxEntity(Entity):
+    """Wrapper for Elmax entities."""
+
+    def __init__(
+        self,
+        panel: PanelEntry,
+        elmax_device: DeviceEndpoint,
+        panel_version: str,
+        coordinator: ElmaxCoordinator,
+    ) -> None:
+        """Construct the object."""
+        self._panel = panel
+        self._device = elmax_device
+        self._panel_version = panel_version
+        self._coordinator = coordinator
+        self._transitory_state = None
+
+    @property
+    def transitory_state(self) -> Any | None:
+        """Return the transitory state for this entity."""
+        return self._transitory_state
+
+    @transitory_state.setter
+    def transitory_state(self, value: Any) -> None:
+        """Set the transitory state value."""
+        self._transitory_state = value
+
+    @property
+    def panel_id(self) -> str:
+        """Retrieve the panel id."""
+        return self._panel.hash
+
+    @property
+    def unique_id(self) -> str | None:
+        """Provide a unique id for this entity."""
+        return self._device.endpoint_id
+
+    @property
+    def name(self) -> str | None:
+        """Return the entity name."""
+        return self._device.name
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return extra attributes."""
+        return {
+            "index": self._device.index,
+            "visible": self._device.visible,
+        }
+
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        return {
+            "identifiers": {(DOMAIN, self._panel.hash)},
+            "name": self._panel.get_name_by_user(
+                self._coordinator.http_client.get_authenticated_username()
+            ),
+            "manufacturer": "Elmax",
+            "model": self._panel_version,
+            "sw_version": self._panel_version,
+        }
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._panel.online
+
+    def _http_data_changed(self) -> None:
+        # Whenever new HTTP data is received from the coordinator we extract the stat of this
+        # device and store it locally for later use
+        device_state = self._coordinator.get_endpoint_state(self._device.endpoint_id)
+        if self._device is None or device_state.__dict__ != self._device.__dict__:
+            # If HTTP data has changed, we need to schedule a forced refresh
+            self._device = device_state
+            self.async_schedule_update_ha_state(force_refresh=True)
+
+        # Reset the transitory state as we did receive a fresh state
+        self._transitory_state = None
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return False
```

Comment:
```suggestion
```

Use `_attr_should_poll = False`

Reply:
Disagree. The leading underscore in a variable stands for "private variables", as per [python convention](_attr_should_poll). Literally, 
```
a name prefixed with an underscore should be treated as a non-public part of the API (whether it is a function, 
a method or a data member). It should be considered an implementation detail and subject to change without
notice.
```
In my opinion overriding a public property (i.e. `should_poll()`) is much better as it complies with the "public" interface offered by the framework, which is expected to be supported on longer terms.

Reply:
https://developers.home-assistant.io/docs/core/entity#entity-class-or-instance-attributes

Reply:
Honestly I don't like this sort of approach with attributes and properties (as it makes the default implementation a bit implicit, despite the second [principle of the zen of python](https://www.python.org/dev/peps/pep-0020/)). 

It is true that the "implicit" \_attr\_<name> form is quicker and might save the developer from writing couple of lines more. However, from the documentation you linked, it is clearly stated the _attr_form is a "shorter" way for achieving the same goal, it's not mandatory: overriding properties is absolutely ok. I am happily paying the effort of writing property getters for the sake of readability.

Reply:
:+1: